data_pipeline/gmail_data/get_gmail_data.py

```diff
@@ -46,14 +52,94 @@ def iter_link_message_thread(service: Resource, user_id: str) -> Iterable[dict]:
 
 
 def get_link_message_thread_ids(service: Resource, user_id: str) -> pd.DataFrame:
-    df_link = pd.DataFrame(iter_link_message_thread(service, user_id))
+    imported_timestamp = get_current_timestamp()
+    df_link = pd.DataFrame(iter_link_message_thread_ids(service, user_id))
+    df_link['user_id'] = user_id
+    df_link['imported_timestamp'] = imported_timestamp
     return df_link
 
 
+@backoff.on_exception(backoff.expo, TimeoutError, max_tries=10)
+def get_one_thread(service: str, user_id: str, thread_id: str) -> pd.DataFrame:
```

Comment:
Here I would still wonder whether it would be beneficial to just dump the data into BigQuery and do the counts via BigQuery views or Data Studio reports? (see https://github.com/elifesciences/data-hub-core-airflow-dags/pull/391#discussion_r592527299)

Reply:
I am not sure about that, maybe tomorrow I can show you the data and we can decide together. 
Because the data here is complex and duplicated many times, if we get as dump the data unnecessarily will be in our DB. If there is not a huge difference with the performance, preferring getting summary information would be easier to manage.
And having a fixed schema it will be difficult to manage if get this nested json dump.

Reply:
Okay, good idea. Let's discuss it later.

Just noting down some benefits I would see with dumping the data as is (i.e. doing more ELT instead of ETL):
- pipeline would be more generic
- it keeps the business logic in one place (bigquery views), although those have admittedly the issue of being more difficult to test
- it will be easier to apply different logic by creating another view, rather than having to retrieve the data again anf filtering it in a different way
- it will provide more data we might be interested later, such as the body text (although it's fair to argue that we can worry about it when it is needed)
- it will make it easier to update as you could just add new messages to a thread and then aggregate in BigQuery (now you would need to retrieve all messages of a thread in order to update the thread summary information)

Regarding the schema, I would be interested to get he auto-schema to work (or at least understand why it wasn't working). Although there may still be an argument to specify it manually.
We could also select the fields as I think you are doing for the labels.